file/src/main/scala/akka/stream/alpakka/file/scaladsl/Directory.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.file.scaladsl
+
+import java.nio.file.attribute.BasicFileAttributes
+import java.nio.file.{Files, Path}
+import java.util.function.BiPredicate
+
+import akka.NotUsed
+import akka.stream.scaladsl.{Source, StreamConverters}
+
+object Directory {
+
+  /**
+   * List all files in the given directory
+   */
+  def ls(directory: Path): Source[Path, NotUsed] = {
+    require(Files.isDirectory(directory), s"Path must be a directory, $directory isn't")
+    StreamConverters.fromJavaStream(() => Files.list(directory))
+  }
+
+  /**
+   * Recursively list files in the given directory and its subdirectories. Listing is done
+   * depth first.
+   */
+  def walk(directory: Path): Source[Path, NotUsed] = {
```

Comment:
This was rather from the not covering every conceivable use case under the sun, just the "normal" ones point of view, especially the filter which could be achieved by composition (with a bit of alloc overhead, but hey this is disk io we are talking about here).

Could you come up with a common use case for depths other than 0, which is covered by ls and all the depth (`find ./`) which is covered by this?